### PR TITLE
[bitnami/contour-operator] Add tests and publishing using VIB

### PR DESCRIPTION
**Pull request by FraPazGal**
Signed-off-by: FraPazGal <fdepaz@vmware.com>

### Description of the change

The main objective of this PR is to publish our Bitnami Contour Operator Helm Chart using VMware Image Builder. In order to do that, several changes are included:

- Adding the asset to the publishing workflow
- Increasing the existing test coverage of the asset by adding Ginkgo tests.

### Benefits

- Ensuring higher quality of the Helm charts.
- Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could show to be potentially flaky. 

### Applicable issues

NA

### Additional information

Tested in https://github.com/FraPazGal/charts/pull/127

### Checklist

- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: .vib/contour-operator/ginkgo/integration_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	}
+	return false
+}
+
+func CheckRequirements() {
+	if *namespace == "" {
+		panic(fmt.Sprintf("The namespace where %s is deployed must be provided. Use the '--namespace' flag", APP_NAME))
+	}
+	if *contourName == "" {
+		panic("The name of the contour resource. Use the '--ingress-name' flag")
```
Comment (joancafom) [inline .vib/contour-operator/ginkgo/integration_suite_test.go:140]:
```suggestion
		panic("The name of the contour resource must be provided. Use the '--contour-crd-name' flag")
```

File: .vib/contour-operator/ginkgo/integration_suite_test.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	}
+	if *ingressName == "" {
+		panic("The resource name of the testing ingress. Use the '--ingress-name' flag")
```
Comment (joancafom) [inline .vib/contour-operator/ginkgo/integration_suite_test.go:143]:
```suggestion
		panic("The resource name of the testing ingress must be provided. Use the '--ingress-name' flag")
```

File: .vib/contour-operator/ginkgo/contour_operator.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+		BeforeEach(func() {
+			controllerPods = getPodsByLabelOrDie(ctx, coreclient, *namespace, "app.kubernetes.io/component=contour-operator")
+			containerLogs = getContainerLogsOrDie(ctx, coreclient, *namespace, controllerPods.Items[0].GetName(), "contour-operator")
+
+			ingress, err = netclient.Ingresses(*namespace).Get(ctx, *ingressName, metav1.GetOptions{})
+			if err != nil {
+				panic(fmt.Sprintf("There was an error retrieving the %q Ingress resource: %q", *ingressName, err))
+			}
+			responseBody = getResponseBodyOrDie(ctx, "http://"+ingress.Status.LoadBalancer.Ingress[0].IP)
```
Comment (joancafom) [inline .vib/contour-operator/ginkgo/contour_operator.go:45]: You should also assert that this IP was indeed managed by the Contour resource and not the cloud K8s platform.

**Review by joancafom — COMMENTED** [review]
Thanks for your PR @FraPazGal ! I left you some comments as a review

File: .vib/contour-operator/ginkgo/contour_operator.go (hunk 3 above)
Comment (FraPazGal) [reply .vib/contour-operator/ginkgo/contour_operator.go]: Added, thanks!

File: .vib/contour-operator/ginkgo/integration_suite_test.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	}
+	return false
+}
+
+func CheckRequirements() {
+	if *namespace == "" {
+		panic(fmt.Sprintf("The namespace where %s is deployed must be provided. Use the '--namespace' flag", APP_NAME))
+	}
+	if *contourName == "" {
+		panic("The name of the contour resource must be provided. Use the '--ingress-name' flag")
```
Comment (joancafom) [inline .vib/contour-operator/ginkgo/integration_suite_test.go:140]:
```suggestion
		panic("The name of the contour resource must be provided. Use the '--contour-crd-name' flag")
```

**Review by joancafom — COMMENTED** [review]
Minor comment

File: .vib/contour-operator/ginkgo/integration_suite_test.go (hunk 4 above)
Comment (FraPazGal) [reply .vib/contour-operator/ginkgo/integration_suite_test.go]: Fixed now, thanks for the catch 😄

**Review by joancafom — DISMISSED** [review]

**Review by joancafom — APPROVED** [review]

**Pull request merged**